### ISSUE TITLE
Allowing single-page mode only (useful for presentations)

### ISSUE THREAD

**Issue by A3EKV** [ahrm/sioyek#52]
As far as I am aware, sioyek has only a continous display mode between pages, and there is no way to display document in a single-page mode which is very useful for presentations. In Zathura pdf viewer, you can achieve that by pressing "F5" key and use the "Page Down" or "Page Up" keys to navigate (which is a partial solution, for details, see this [page](https://unix.stackexchange.com/questions/73281/change-zathuras-display-mode-to-single-page-only)).

Is it possible to have at least a similar behaviour for presentations mode?

Thank you

**Comment by ahrm** [comment]
I added a simple presentation mode in 4f69f0ab299cf200e21c074fd89b2f58f0be92f8 . You presss `<f5>` by default to enter presentation mode (single page mode) and then you can use `next_page` and `prev_page` (`ctrl+pagedown` and `ctrl+pageup` by default) to move between pages.

**Comment by A3EKV** [comment]
> I added a simple presentation mode in [4f69f0a](https://github.com/ahrm/sioyek/commit/4f69f0ab299cf200e21c074fd89b2f58f0be92f8) . You presss `<f5>` by default to enter presentation mode (single page mode) and then you can use `next_page` and `prev_page` (`ctrl+pagedown` and `ctrl+pageup` by default) to move between pages.

I have cloned the repository again and compiled it under Linux but apparently it doesn't work. When I press F5 it does nothing.
I have double checked the keys.config file it has the entry to toggle the presentation mode.

Any ideas?

**Comment by ahrm** [comment]
> > I added a simple presentation mode in [4f69f0a](https://github.com/ahrm/sioyek/commit/4f69f0ab299cf200e21c074fd89b2f58f0be92f8) . You presss `<f5>` by default to enter presentation mode (single page mode) and then you can use `next_page` and `prev_page` (`ctrl+pagedown` and `ctrl+pageup` by default) to move between pages.
> 
> I have cloned the repository again and compiled it under Linux but apparently it doesn't work. When I press F5 it does nothing.
> I have double checked the keys.config file it has the entry to toggle the presentation mode.
> 
> Any ideas?

Are you sure you have checked the correct `keys.config` file? Is it the one which is printed on console when you launch sioyek?

**Comment by A3EKV** [comment]
Yes, when Sioyek starts it prints some default paths:
```
...
default_keys_path: /home/ae3/coding/git/sioyek/build/keys.config
```
And yes, that's the config file I am editing

**Comment by ahrm** [comment]
> Yes, when Sioyek starts it prints some default paths:
> 
> ```
> ...
> default_keys_path: /home/ae3/coding/git/sioyek/build/keys.config
> ```
> 
> And yes, that's the config file I am editing

Also what do you expect should happen when you press `<f5>`? If you already have a page larger than screen on the page then nothing visible happens. It is only visible when you are zoomed out a lot.

**Comment by A3EKV** [comment]
Thank you.
I feel that this feature is far from being perfect, maybe you can compare that with zathura, or any other viewer, such as evince.
In my experience with those PDF viewers, it doesn't matter at what level of zooming the document is, when you press F5 you get a full screen with ONLY one page displayed. (Only one page can be displayed at a time, for example you can't see a part from page 4 and a part from page5).

**Comment by ahrm** [comment]
> Thank you.
> I feel that this feature is far from being perfect, maybe you can compare that with zathura, or any other viewer, such as evince.
> In my experience with those PDF viewers, it doesn't matter at what level of zooming the document is, when you press F5 you get a full screen with ONLY one page displayed. (Only one page can be displayed at a time, for example you can't see a part from page 4 and a part from page5).

Well yes, that is how this feature works here too. Is it not still working for you?

**Comment by A3EKV** [comment]
I have to manually adjust the size of each page to fit, or when I press F10 it some parts of the page are missing.

**Comment by ahrm** [comment]
So what you want is to automatically adjust screen size when entering presentation mode so that it fits the window width/height?

**Comment by A3EKV** [comment]
Yes, that's what the presentation mode is supposed to do. I have tested that with Okular, evince, and zathura all of them do the same and automatically enter a full screen mode.

**Comment by ahrm** [comment]
I have added this feature in 3ec297635ec2726284c043e976e899cf26f48c61 . Note that while in presentation mode, normal moving keys (mouse scroll, arrow keys, etc. ) don't work and you must use `next_page` and `prev_page` (`ctrl+pagedown` and `ctrl+pageup`) for navigation.

**Comment by A3EKV** [comment]
Nice!
However:
 * There is no indication that you are in the presentation or in normal mode. there should be some indication such as a message or it toggle the full screen mode.
 * If you start the presentation mode in a random page, say page 6, the other navigation keys are disabled as expected but the `Pageup` key still works and you can navigate up (the `Pagedown` is disabled as expected).

**Comment by ahrm** [comment]
> Nice!
> However:
> 
> * There is no indication that you are in the presentation or in normal mode. there should be some indication such as a message or it toggle the full screen mode.
> * If you start the presentation mode in a random page, say page 6, the other navigation keys are disabled as expected but the `Pageup` key still works and you can navigate up (the `Pagedown` is disabled as expected).

Fixed in bd7fab0856272d166cf82a7d7087d2b393376285 . Now instead of being disabled, normal movement keys act as next page and previous page while we are in presentation mode.
